src/pubd/repository.rs

```diff
@@ -395,13 +403,39 @@ impl RsyncdStore {
         old_dir.push("old");
 
         if current_dir.exists() {
-            fs::rename(&current_dir, &old_dir)?;
+            fs::rename(&current_dir, &old_dir).map_err(|e| {
+                KrillIoError::new(
+                    format!(
+                        "Could not rename rsync dir from '{}' to '{}' while publishing",
+                        current_dir.to_string_lossy(),
+                        old_dir.to_string_lossy()
+                    ),
+                    e,
+                )
+            })?;
         }
 
-        fs::rename(&new_dir, &current_dir)?;
+        fs::rename(&new_dir, &current_dir).map_err(|e| {
+            KrillIoError::new(
+                format!(
+                    "Could not rename rsync dir from '{}' to '{}' while publishing",
+                    new_dir.to_string_lossy(),
+                    current_dir.to_string_lossy()
+                ),
+                e,
+            )
+        })?;
 
         if old_dir.exists() {
-            fs::remove_dir_all(&old_dir)?;
+            fs::remove_dir_all(&old_dir).map_err(|e| {
+                KrillIoError::new(
+                    format!(
+                        "Could not clean up old rsync dir '{}' while publishing",
```

Comment:
Above you say remove, here you say cleanup. Cleanup to me could mean reorganize or remove clutter, it doesn't necessarily mean delete. Perhaps reword this?

Reply:
ack